test/runtests.py

```diff
@@ -7,114 +7,208 @@
+                print("Expected Output:")
+                print("----------------------------")
+                print(lst_expected[i])
+                print("----------------------------")
                 break
 
-    print "exit code:", exit_code
-    print "\nFailed!"
-    sys.exit(exit_code)
+    print("exit code: {}".format(exit_code))
+    test_result.log(filename, fail=True)
 
-def test_path(folder, is_file):
-    files=['']
-    if is_file == False:
-        print "Testing ->", os.path.basename(folder)
-        files = os.listdir(folder)
+def test_path(path):
+    if os.path.isfile(path):
+        folder, file = os.path.dirname(path), os.path.basename(path)
     else:
-        files[0] = folder
-    
-    for js_file in files:
-        if is_file or os.path.splitext(js_file)[1] == '.js':
-            js_file = os.path.join(folder, js_file)
-            js_output = ""
-
-            if not os.path.isfile(js_file):
-                print "Javascript file doesn't exist (" + js_file + ")"
-                sys.exit(1)
-
-            p = SP.Popen([ch_path, js_file], stdout=SP.PIPE, stderr=SP.STDOUT, close_fds=True)
-            js_output = p.communicate()[0].replace('\r','')
-            exit_code = p.wait()
-
-            if exit_code != 0:
-                show_failed(js_file, js_output, exit_code, None)
-            else: #compare outputs
-                baseline = os.path.splitext(js_file)[0] + '.baseline'
-                baseline = os.path.join(folder, baseline)
-                if os.path.isfile(baseline):
-                    expected_output = None
-                    with open(baseline, 'r') as bs_file:
-                        expected_output = bs_file.read().replace('\r', '')
-                    # todo: compare line by line and use/implement wild cards support
-                    # todo: by default we discard line endings (xplat), make this optional
-                    if expected_output.replace('\n', '') != js_output.replace('\n', ''):
-                        show_failed(js_file, js_output, exit_code, expected_output)
-
-            if not is_file:
-                print "\tPassed ->", os.path.basename(js_file)
-
-is_file = len(test_dirs) == 1 and os.path.splitext(test_dirs[0])[1] == '.js'
-
-for folder in test_dirs:
-    full_path = os.path.join(test_root, folder)
-    if os.path.isdir(full_path) or is_file:
-        test_path(full_path, is_file)
-
-print 'Success!'
+        folder, file = path, None
+
+    tests = load_tests(folder, file)
+    if len(tests) == 0:
+        return
+
+    print("Testing -> " + os.path.basename(folder))
+    for test in tests:
+        test_one(folder, test)
+
+def test_one(folder, test):
+    js_file = os.path.join(folder, test['files'])
+    js_output = ""
+
+    flags = test.get('compile-flags')
+    cmd = [binary, '-WERExceptionSupport', '-ExtendedErrorStackForTestHost'] \
+          + (flags.split() if flags else []) \
+          + [js_file]
```

Comment:
I think if you put the pluses at the end of the line it will look to the next line for the rest of the expression so you wouldn't need the `\` at the ends of the lines.